command/jsonstate/state.go

```diff
@@ -266,39 +269,78 @@ func marshalResources(resources map[string]*states.Resource, schemas *terraform.
 			}
 
 			if r.EachMode != states.NoEach {
-				resource.Index = k
+				current.Index = k
 			}
 
 			schema, _ := schemas.ResourceTypeConfig(
 				r.ProviderConfig.ProviderConfig.Type,
 				r.Addr.Mode,
 				r.Addr.Type,
 			)
-			resource.SchemaVersion = ri.Current.SchemaVersion
 
-			if schema == nil {
-				return nil, fmt.Errorf("no schema found for %s", r.Addr.String())
-			}
-			riObj, err := ri.Current.Decode(schema.ImpliedType())
-			if err != nil {
-				return nil, err
-			}
+			// It is possible that the only instance is deposed
+			if ri.Current != nil {
+				current.SchemaVersion = ri.Current.SchemaVersion
 
-			resource.AttributeValues = marshalAttributeValues(riObj.Value, schema)
+				if schema == nil {
+					return nil, fmt.Errorf("no schema found for %s", r.Addr.String())
+				}
+				riObj, err := ri.Current.Decode(schema.ImpliedType())
+				if err != nil {
+					return nil, err
+				}
+
+				current.AttributeValues = marshalAttributeValues(riObj.Value, schema)
 
-			if len(riObj.Dependencies) > 0 {
-				dependencies := make([]string, len(riObj.Dependencies))
-				for i, v := range riObj.Dependencies {
-					dependencies[i] = v.String()
+				if len(riObj.Dependencies) > 0 {
+					dependencies := make([]string, len(riObj.Dependencies))
+					for i, v := range riObj.Dependencies {
+						dependencies[i] = v.String()
+					}
+					current.DependsOn = dependencies
 				}
-				resource.DependsOn = dependencies
+
+				if riObj.Status == states.ObjectTainted {
+					current.Tainted = true
+				}
+				ret = append(ret, current)
 			}
 
-			if riObj.Status == states.ObjectTainted {
-				resource.Tainted = true
+			if ri.Deposed != nil {
```

Comment:
You don't the `if` here if you're going to rage over the map to reduce indentation.

Reply:
🤦‍♀ thank you, great catch! That was leftover from an earlier implementation. Erp.